crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_list_call.rs

```diff
@@ -45,7 +45,7 @@ pub fn unnecessary_list_call(checker: &mut Checker, expr: &Expr, func: &Expr, ar
     if !checker.ctx.is_builtin("list") {
         return;
     }
-    if !matches!(argument, ExprKind::ListComp { .. }) {
+    if !matches!(argument, ExprKind::ListComp(_)) {
```

Comment:
Idea: It would be nice to have `argument.is_<node>()` methods on unions so that we could simplify this to 

```rust
if !argument.is_list_comp() {
	...
}
```

Reply:
Yes, definitely. Actually that's also done in my local repository a few hours ago xD